tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -136,6 +149,104 @@ ZTEST_F(ascs_test_suite, test_sink_ase_read_state_idle)
 	zassert_equal(0x00, hdr.ase_state, "unexpected ASE_State 0x%02x", hdr.ase_state);
 }
 
+ZTEST_F(ascs_test_suite, test_cb_register_without_ascs_registered)
+{
+	int err;
+
+	/* Unregister ASCS, as its registered through setup */
+	bt_bap_unicast_server_unregister();
+
+	err = bt_bap_unicast_server_register_cb(&mock_bap_unicast_server_cb);
+
+	zassert_equal(err, -ENOTSUP, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_with_null_param)
+{
+	int err;
+
+	/* Unregister ASCS, as its registered through setup */
+	bt_bap_unicast_server_unregister();
+
+	err = bt_bap_unicast_server_register(NULL);
+
+	zassert_equal(err, -EINVAL, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_twice)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT
+	};
+
+	/* Setup already registered once, so calling once here should be sufficient */
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, -EALREADY, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_too_many_sinks)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT + 1,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT
+	};
+
+	/* Unregister ASCS, as its registered through setup */
+	bt_bap_unicast_server_unregister();
+
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, -EINVAL, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_too_many_sources)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT + 1
+	};
```

Comment:
Should we have a test where both are set to 0?

Reply:
We should, and we should also check that if that happens, then we should not register, as its mandatory to have at least one ASE. That is not being checked today,